### PR TITLE
Restart button + Clear code button

### PR DESCRIPTION
**Pull request by m-lin1**
For issues #83 and #81

### CODE REVIEW

File: Scripts/Level_Components/IDE_Elements/IfBlock.gd
Comment (he-larious) [inline Scripts/Level_Components/IDE_Elements/IfBlock.gd:18]: Did you mean to comment this out?

File: Scripts/Level_Components/IDE_Elements/LoopBlock.gd
Comment (he-larious) [inline Scripts/Level_Components/IDE_Elements/LoopBlock.gd:36]: Ye once again why is this commented out now?

File: Scripts/Singletons/PuzzleElementPlacement.gd
Comment (he-larious) [inline Scripts/Singletons/PuzzleElementPlacement.gd:297]: Oh wait nvm ignore my other 2 comments I see why you commented it out now

**Review by he-larious — COMMENTED** [review]
Everything looks good to me but before I approve I wanna see if I can get the focus highlight to not go away when the player presses clear code/restart. I also wanna play test the loops more.

**Review by he-larious — APPROVED** [review]
Wait for additional comments to PEP.gd + review of function highlight feature before merging.

**Comment by m-lin1** [comment]
> Wait for additional comments to PEP.gd + review of function highlight feature before merging.

The function highlight feature works + Added the comments. I can either merge rn or merge after either you or someone else double checks that the Dialogue progress checks post-Restart work.

File: Scripts/Singletons/PuzzleElementPlacement.gd
Comment (he-larious) [inline Scripts/Singletons/PuzzleElementPlacement.gd:354]: I'm crying you really said let me hard code this real quick. This will also need to be updated with the loop and if blocks later on

Comment (he-larious) [reply Scripts/Singletons/PuzzleElementPlacement.gd:354]: We should allow up to 3 func blocks, 3 if-else blocks, and 3 loop blocks. Also is there a better way to do this or nah

Comment (m-lin1) [reply Scripts/Singletons/PuzzleElementPlacement.gd:360]: mi hoy minoy

**Review by he-larious — APPROVED** [review]
You finally turned your hardcode into softcode ong

**Pull request merged**
